### PR TITLE
Fix `darwin` flag when cross-compiling from Linux to MacOS

### PR DESCRIPTION
**Pull request by benradf**
In response to [this comment](https://github.com/tweag/rules_nixpkgs/pull/402#discussion_r1316920564) from @jcpetruzza.

### CODE REVIEW

File: toolchains/cc/cc.bzl [hunk 1]
```diff
@@ -122,8 +122,8 @@ def _parse_cc_toolchain_info(content, filename):
 
 def _nixpkgs_cc_toolchain_config_impl(repository_ctx):
     host_cpu = get_cpu_value(repository_ctx)
-    cross_cpu = repository_ctx.attr.cross_cpu or host_cpu
-    darwin = (host_cpu == "darwin" or host_cpu == "darwin_arm64") and cross_cpu == host_cpu
+    cpu_value = repository_ctx.attr.cross_cpu or host_cpu
+    darwin = cpu_value == "darwin" or cpu_value == "darwin_arm64"
```
Comment (DolceTriade) [inline toolchains/cc/cc.bzl:126]: Yes, I think this is correct and will ensure the correct version of _cc_wrapper is selected.

**Review by DolceTriade — APPROVED** [review]

File: examples/toolchains/cc_cross_osx_to_linux_amd64/README.md [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
 # Usage
 
-To run the example with Nix, issue the following command:
+To build the image with Nix, issue the following command:
+```
+nix-shell --command 'bazel build --config=cross :hello_image_tarball'
 ```
-nix-shell --command 'bazel run --config=cross :hello'
+
+You can then do load the image into Docker with:
```
Comment (aherrmann) [inline examples/toolchains/cc_cross_osx_to_linux_amd64/README.md:13]: It's also possible to just load the image with `bazel run`, [see here](https://github.com/bazelbuild/rules_docker/tree/master#using-with-docker-locally).

**Review by aherrmann — APPROVED** [review]
Thank you!

File: examples/toolchains/cc_cross_osx_to_linux_amd64/README.md (hunk 2 above)
Comment (benradf) [reply examples/toolchains/cc_cross_osx_to_linux_amd64/README.md:13]: I was testing on our `build02` machine, which doesn't have Docker installed, but I think it's fair to assume Docker is there for the README. Will update.

**Pull request merged**